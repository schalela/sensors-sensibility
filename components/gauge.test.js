import { render } from 'react-testing-library';
import Gauge from './gauge';

describe('Gauge', () => {
  it('should render', () => {
    const container = render(<Gauge
      icon={'/static/ambient_temperature.png'}
      startColor={'green'}
      endColor={'red'}
      maxValue={50}
      value={10}
    />);
    expect(container).toMatchSnapshot();
  });
});
