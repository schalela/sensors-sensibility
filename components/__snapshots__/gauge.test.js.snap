// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gauge should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        src="/static/ambient_temperature.png"
      />
      <div>
        <svg
          class="speedometer"
          height="110"
          width="110"
        >
          <g
            class="arc"
            transform="translate(55,55)"
          >
            <path
              class="speedo-segment"
              d="M-35,-4.2862637970157365e-15A35,35,0,0,1,-34.72401454600672,-4.386663174750659L-22.818638130232987,-2.882664371979004A23,23,0,0,0,-23,-2.8166876380389124e-15Z"
              fill="rgb(0, 0, 0)"
            />
            <path
              class="speedo-segment"
              d="M-34.72401454600672,-4.386663174750659A35,35,0,0,1,-33.900410639502084,-8.704146050769934L-22.27741270595851,-5.71986740479167A23,23,0,0,0,-22.818638130232987,-2.882664371979004Z"
              fill="rgb(10, 10, 10)"
            />
            <path
              class="speedo-segment"
              d="M-33.900410639502084,-8.704146050769934A35,35,0,0,1,-32.5421770060888,-12.884359343963721L-21.384859175429785,-8.466864711747588A23,23,0,0,0,-22.27741270595851,-5.71986740479167Z"
              fill="rgb(20, 20, 20)"
            />
            <path
              class="speedo-segment"
              d="M-32.5421770060888,-12.884359343963721A35,35,0,0,1,-30.670733801535224,-16.861378593560033L-20.155053641008863,-11.08033450433945A23,23,0,0,0,-21.384859175429785,-8.466864711747588Z"
              fill="rgb(31, 31, 31)"
            />
            <path
              class="speedo-segment"
              d="M-30.670733801535224,-16.861378593560033A35,35,0,0,1,-28.315594803123158,-20.572483830236564L-18.60739087062379,-13.519060802726885A23,23,0,0,0,-20.155053641008863,-11.08033450433945Z"
              fill="rgb(41, 41, 41)"
            />
            <path
              class="speedo-segment"
              d="M-28.315594803123158,-20.572483830236564A35,35,0,0,1,-25.513901959749397,-23.95914870750411L-16.76627843069246,-15.744583436359843A23,23,0,0,0,-18.60739087062379,-13.519060802726885Z"
              fill="rgb(51, 51, 51)"
            />
            <path
              class="speedo-segment"
              d="M-25.513901959749397,-23.95914870750411A35,35,0,0,1,-22.309839641204128,-26.96796349715263L-14.660751764219857,-17.72180458384316A23,23,0,0,0,-16.76627843069246,-15.744583436359843Z"
              fill="rgb(61, 61, 61)"
            />
            <path
              class="speedo-segment"
              d="M-22.309839641204128,-26.96796349715263A35,35,0,0,1,-18.753937824264874,-29.551477392570533L-12.324016284516919,-19.41954228654635A23,23,0,0,0,-14.660751764219857,-17.72180458384316Z"
              fill="rgb(71, 71, 71)"
            />
            <path
              class="speedo-segment"
              d="M-18.753937824264874,-29.551477392570533A35,35,0,0,1,-14.902275204777546,-31.66894683631068L-9.792923705996673,-20.811022206718448A23,23,0,0,0,-12.324016284516919,-19.41954228654635Z"
              fill="rgb(82, 82, 82)"
            />
            <path
              class="speedo-segment"
              d="M-14.902275204777546,-31.66894683631068A35,35,0,0,1,-10.815594803123156,-33.28697807033038L-7.107390870623789,-21.874299874788534A23,23,0,0,0,-9.792923705996673,-20.811022206718448Z"
              fill="rgb(92, 92, 92)"
            />
            <path
              class="speedo-segment"
              d="M-10.815594803123156,-33.28697807033038A35,35,0,0,1,-6.558346010500361,-34.3800537755041L-4.309770235471666,-22.59260676675984A23,23,0,0,0,-7.107390870623789,-21.874299874788534Z"
              fill="rgb(102, 102, 102)"
            />
            <path
              class="speedo-segment"
              d="M-6.558346010500361,-34.3800537755041A35,35,0,0,1,-2.1976681835259693,-34.930935494989505L-1.4441819491742083,-22.954614753850247A23,23,0,0,0,-4.309770235471666,-22.59260676675984Z"
              fill="rgb(112, 112, 112)"
            />
            <path
              class="speedo-segment"
              d="M-2.1976681835259693,-34.930935494989505A35,35,0,0,1,2.1976681835259733,-34.930935494989505L1.4441819491742112,-22.954614753850247A23,23,0,0,0,-1.4441819491742083,-22.954614753850247Z"
              fill="rgb(122, 122, 122)"
            />
            <path
              class="speedo-segment"
              d="M2.1976681835259733,-34.930935494989505A35,35,0,0,1,6.558346010500372,-34.3800537755041L4.309770235471674,-22.59260676675984A23,23,0,0,0,1.4441819491742112,-22.954614753850247Z"
              fill="rgb(133, 133, 133)"
            />
            <path
              class="speedo-segment"
              d="M6.558346010500372,-34.3800537755041A35,35,0,0,1,10.815594803123162,-33.28697807033037L7.107390870623791,-21.87429987478853A23,23,0,0,0,4.309770235471674,-22.59260676675984Z"
              fill="rgb(143, 143, 143)"
            />
            <path
              class="speedo-segment"
              d="M10.815594803123162,-33.28697807033037A35,35,0,0,1,14.902275204777549,-31.66894683631068L9.792923705996674,-20.811022206718448A23,23,0,0,0,7.107390870623791,-21.87429987478853Z"
              fill="rgb(153, 153, 153)"
            />
            <path
              class="speedo-segment"
              d="M14.902275204777549,-31.66894683631068A35,35,0,0,1,18.753937824264888,-29.551477392570522L12.324016284516926,-19.419542286546346A23,23,0,0,0,9.792923705996674,-20.811022206718448Z"
              fill="rgb(163, 163, 163)"
            />
            <path
              class="speedo-segment"
              d="M18.753937824264888,-29.551477392570522A35,35,0,0,1,22.30983964120414,-26.967963497152624L14.660751764219862,-17.721804583843152A23,23,0,0,0,12.324016284516926,-19.419542286546346Z"
              fill="rgb(173, 173, 173)"
            />
            <path
              class="speedo-segment"
              d="M22.30983964120414,-26.967963497152624A35,35,0,0,1,25.513901959749408,-23.9591487075041L16.76627843069247,-15.744583436359836A23,23,0,0,0,14.660751764219862,-17.721804583843152Z"
              fill="rgb(184, 184, 184)"
            />
            <path
              class="speedo-segment"
              d="M25.513901959749408,-23.9591487075041A35,35,0,0,1,28.31559480312316,-20.57248383023656L18.607390870623792,-13.519060802726882A23,23,0,0,0,16.76627843069247,-15.744583436359836Z"
              fill="rgb(194, 194, 194)"
            />
            <path
              class="speedo-segment"
              d="M28.31559480312316,-20.57248383023656A35,35,0,0,1,30.67073380153522,-16.86137859356004L20.15505364100886,-11.080334504339456A23,23,0,0,0,18.607390870623792,-13.519060802726882Z"
              fill="rgb(204, 204, 204)"
            />
            <path
              class="speedo-segment"
              d="M30.67073380153522,-16.86137859356004A35,35,0,0,1,32.5421770060888,-12.884359343963723L21.384859175429785,-8.46686471174759A23,23,0,0,0,20.15505364100886,-11.080334504339456Z"
              fill="rgb(214, 214, 214)"
            />
            <path
              class="speedo-segment"
              d="M32.5421770060888,-12.884359343963723A35,35,0,0,1,33.900410639502084,-8.704146050769921L22.277412705958515,-5.719867404791663A23,23,0,0,0,21.384859175429785,-8.46686471174759Z"
              fill="rgb(224, 224, 224)"
            />
            <path
              class="speedo-segment"
              d="M33.900410639502084,-8.704146050769921A35,35,0,0,1,34.72401454600672,-4.386663174750662L22.818638130232987,-2.882664371979007A23,23,0,0,0,22.277412705958515,-5.719867404791663Z"
              fill="rgb(235, 235, 235)"
            />
            <path
              class="speedo-segment"
              d="M34.72401454600672,-4.386663174750662A35,35,0,0,1,35,0L23,0A23,23,0,0,0,22.818638130232987,-2.882664371979007Z"
              fill="rgb(245, 245, 245)"
            />
          </g>
          <g
            class="label"
            transform="translate(55,55)"
          >
            <text
              class="segment-value"
              style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
              transform="rotate(-90) translate(0,-45)"
            >
              0
            </text>
            <text
              class="segment-value"
              style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
              transform="rotate(-54) translate(0,-45)"
            >
              10
            </text>
            <text
              class="segment-value"
              style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
              transform="rotate(-18) translate(0,-45)"
            >
              20
            </text>
            <text
              class="segment-value"
              style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
              transform="rotate(18) translate(0,-45)"
            >
              30
            </text>
            <text
              class="segment-value"
              style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
              transform="rotate(54) translate(0,-45)"
            >
              40
            </text>
            <text
              class="segment-value"
              style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
              transform="rotate(90) translate(0,-45)"
            >
              50
            </text>
          </g>
          <g
            transform="translate(55,55)"
          >
            <text
              class="current-value"
              style="font-size: 16px; font-weight: bold; fill: #666;"
              text-anchor="middle"
              y="23"
            >
              10
            </text>
          </g>
          <g
            class="pointer"
            style="fill: white;"
            transform="translate(55,55)"
          >
            <path
              d="M5,0C3.333333333333333,-19.5,1.6666666666666667,-39,0,-39C-1.6666666666666667,-39,-3.333333333333333,0,-5,0C-3.333333333333333,0,-1.6666666666666667,5,0,5C1.6666666666666667,5,3.333333333333333,2.5,5,0"
              transform="rotate(-90)"
            />
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <img
      src="/static/ambient_temperature.png"
    />
    <div>
      <svg
        class="speedometer"
        height="110"
        width="110"
      >
        <g
          class="arc"
          transform="translate(55,55)"
        >
          <path
            class="speedo-segment"
            d="M-35,-4.2862637970157365e-15A35,35,0,0,1,-34.72401454600672,-4.386663174750659L-22.818638130232987,-2.882664371979004A23,23,0,0,0,-23,-2.8166876380389124e-15Z"
            fill="rgb(0, 0, 0)"
          />
          <path
            class="speedo-segment"
            d="M-34.72401454600672,-4.386663174750659A35,35,0,0,1,-33.900410639502084,-8.704146050769934L-22.27741270595851,-5.71986740479167A23,23,0,0,0,-22.818638130232987,-2.882664371979004Z"
            fill="rgb(10, 10, 10)"
          />
          <path
            class="speedo-segment"
            d="M-33.900410639502084,-8.704146050769934A35,35,0,0,1,-32.5421770060888,-12.884359343963721L-21.384859175429785,-8.466864711747588A23,23,0,0,0,-22.27741270595851,-5.71986740479167Z"
            fill="rgb(20, 20, 20)"
          />
          <path
            class="speedo-segment"
            d="M-32.5421770060888,-12.884359343963721A35,35,0,0,1,-30.670733801535224,-16.861378593560033L-20.155053641008863,-11.08033450433945A23,23,0,0,0,-21.384859175429785,-8.466864711747588Z"
            fill="rgb(31, 31, 31)"
          />
          <path
            class="speedo-segment"
            d="M-30.670733801535224,-16.861378593560033A35,35,0,0,1,-28.315594803123158,-20.572483830236564L-18.60739087062379,-13.519060802726885A23,23,0,0,0,-20.155053641008863,-11.08033450433945Z"
            fill="rgb(41, 41, 41)"
          />
          <path
            class="speedo-segment"
            d="M-28.315594803123158,-20.572483830236564A35,35,0,0,1,-25.513901959749397,-23.95914870750411L-16.76627843069246,-15.744583436359843A23,23,0,0,0,-18.60739087062379,-13.519060802726885Z"
            fill="rgb(51, 51, 51)"
          />
          <path
            class="speedo-segment"
            d="M-25.513901959749397,-23.95914870750411A35,35,0,0,1,-22.309839641204128,-26.96796349715263L-14.660751764219857,-17.72180458384316A23,23,0,0,0,-16.76627843069246,-15.744583436359843Z"
            fill="rgb(61, 61, 61)"
          />
          <path
            class="speedo-segment"
            d="M-22.309839641204128,-26.96796349715263A35,35,0,0,1,-18.753937824264874,-29.551477392570533L-12.324016284516919,-19.41954228654635A23,23,0,0,0,-14.660751764219857,-17.72180458384316Z"
            fill="rgb(71, 71, 71)"
          />
          <path
            class="speedo-segment"
            d="M-18.753937824264874,-29.551477392570533A35,35,0,0,1,-14.902275204777546,-31.66894683631068L-9.792923705996673,-20.811022206718448A23,23,0,0,0,-12.324016284516919,-19.41954228654635Z"
            fill="rgb(82, 82, 82)"
          />
          <path
            class="speedo-segment"
            d="M-14.902275204777546,-31.66894683631068A35,35,0,0,1,-10.815594803123156,-33.28697807033038L-7.107390870623789,-21.874299874788534A23,23,0,0,0,-9.792923705996673,-20.811022206718448Z"
            fill="rgb(92, 92, 92)"
          />
          <path
            class="speedo-segment"
            d="M-10.815594803123156,-33.28697807033038A35,35,0,0,1,-6.558346010500361,-34.3800537755041L-4.309770235471666,-22.59260676675984A23,23,0,0,0,-7.107390870623789,-21.874299874788534Z"
            fill="rgb(102, 102, 102)"
          />
          <path
            class="speedo-segment"
            d="M-6.558346010500361,-34.3800537755041A35,35,0,0,1,-2.1976681835259693,-34.930935494989505L-1.4441819491742083,-22.954614753850247A23,23,0,0,0,-4.309770235471666,-22.59260676675984Z"
            fill="rgb(112, 112, 112)"
          />
          <path
            class="speedo-segment"
            d="M-2.1976681835259693,-34.930935494989505A35,35,0,0,1,2.1976681835259733,-34.930935494989505L1.4441819491742112,-22.954614753850247A23,23,0,0,0,-1.4441819491742083,-22.954614753850247Z"
            fill="rgb(122, 122, 122)"
          />
          <path
            class="speedo-segment"
            d="M2.1976681835259733,-34.930935494989505A35,35,0,0,1,6.558346010500372,-34.3800537755041L4.309770235471674,-22.59260676675984A23,23,0,0,0,1.4441819491742112,-22.954614753850247Z"
            fill="rgb(133, 133, 133)"
          />
          <path
            class="speedo-segment"
            d="M6.558346010500372,-34.3800537755041A35,35,0,0,1,10.815594803123162,-33.28697807033037L7.107390870623791,-21.87429987478853A23,23,0,0,0,4.309770235471674,-22.59260676675984Z"
            fill="rgb(143, 143, 143)"
          />
          <path
            class="speedo-segment"
            d="M10.815594803123162,-33.28697807033037A35,35,0,0,1,14.902275204777549,-31.66894683631068L9.792923705996674,-20.811022206718448A23,23,0,0,0,7.107390870623791,-21.87429987478853Z"
            fill="rgb(153, 153, 153)"
          />
          <path
            class="speedo-segment"
            d="M14.902275204777549,-31.66894683631068A35,35,0,0,1,18.753937824264888,-29.551477392570522L12.324016284516926,-19.419542286546346A23,23,0,0,0,9.792923705996674,-20.811022206718448Z"
            fill="rgb(163, 163, 163)"
          />
          <path
            class="speedo-segment"
            d="M18.753937824264888,-29.551477392570522A35,35,0,0,1,22.30983964120414,-26.967963497152624L14.660751764219862,-17.721804583843152A23,23,0,0,0,12.324016284516926,-19.419542286546346Z"
            fill="rgb(173, 173, 173)"
          />
          <path
            class="speedo-segment"
            d="M22.30983964120414,-26.967963497152624A35,35,0,0,1,25.513901959749408,-23.9591487075041L16.76627843069247,-15.744583436359836A23,23,0,0,0,14.660751764219862,-17.721804583843152Z"
            fill="rgb(184, 184, 184)"
          />
          <path
            class="speedo-segment"
            d="M25.513901959749408,-23.9591487075041A35,35,0,0,1,28.31559480312316,-20.57248383023656L18.607390870623792,-13.519060802726882A23,23,0,0,0,16.76627843069247,-15.744583436359836Z"
            fill="rgb(194, 194, 194)"
          />
          <path
            class="speedo-segment"
            d="M28.31559480312316,-20.57248383023656A35,35,0,0,1,30.67073380153522,-16.86137859356004L20.15505364100886,-11.080334504339456A23,23,0,0,0,18.607390870623792,-13.519060802726882Z"
            fill="rgb(204, 204, 204)"
          />
          <path
            class="speedo-segment"
            d="M30.67073380153522,-16.86137859356004A35,35,0,0,1,32.5421770060888,-12.884359343963723L21.384859175429785,-8.46686471174759A23,23,0,0,0,20.15505364100886,-11.080334504339456Z"
            fill="rgb(214, 214, 214)"
          />
          <path
            class="speedo-segment"
            d="M32.5421770060888,-12.884359343963723A35,35,0,0,1,33.900410639502084,-8.704146050769921L22.277412705958515,-5.719867404791663A23,23,0,0,0,21.384859175429785,-8.46686471174759Z"
            fill="rgb(224, 224, 224)"
          />
          <path
            class="speedo-segment"
            d="M33.900410639502084,-8.704146050769921A35,35,0,0,1,34.72401454600672,-4.386663174750662L22.818638130232987,-2.882664371979007A23,23,0,0,0,22.277412705958515,-5.719867404791663Z"
            fill="rgb(235, 235, 235)"
          />
          <path
            class="speedo-segment"
            d="M34.72401454600672,-4.386663174750662A35,35,0,0,1,35,0L23,0A23,23,0,0,0,22.818638130232987,-2.882664371979007Z"
            fill="rgb(245, 245, 245)"
          />
        </g>
        <g
          class="label"
          transform="translate(55,55)"
        >
          <text
            class="segment-value"
            style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
            transform="rotate(-90) translate(0,-45)"
          >
            0
          </text>
          <text
            class="segment-value"
            style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
            transform="rotate(-54) translate(0,-45)"
          >
            10
          </text>
          <text
            class="segment-value"
            style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
            transform="rotate(-18) translate(0,-45)"
          >
            20
          </text>
          <text
            class="segment-value"
            style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
            transform="rotate(18) translate(0,-45)"
          >
            30
          </text>
          <text
            class="segment-value"
            style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
            transform="rotate(54) translate(0,-45)"
          >
            40
          </text>
          <text
            class="segment-value"
            style="text-anchor: middle; font-size: 14px; font-weight: bold; fill: #666;"
            transform="rotate(90) translate(0,-45)"
          >
            50
          </text>
        </g>
        <g
          transform="translate(55,55)"
        >
          <text
            class="current-value"
            style="font-size: 16px; font-weight: bold; fill: #666;"
            text-anchor="middle"
            y="23"
          >
            10
          </text>
        </g>
        <g
          class="pointer"
          style="fill: white;"
          transform="translate(55,55)"
        >
          <path
            d="M5,0C3.333333333333333,-19.5,1.6666666666666667,-39,0,-39C-1.6666666666666667,-39,-3.333333333333333,0,-5,0C-3.333333333333333,0,-1.6666666666666667,5,0,5C1.6666666666666667,5,3.333333333333333,2.5,5,0"
            transform="rotate(-90)"
          />
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
